pulp_service/pulp_service/app/tasks/ansible_log_parser.py

```diff
@@ -0,0 +1,196 @@
+import re
+import copy
+import logging
+import aiohttp
+from asgiref.sync import sync_to_async
+from pulp_service.app.models import AnsibleLogReport  # Add this import
+from pulpcore.plugin.tasking import dispatch
+
+_logger = logging.getLogger(__name__)
+
+# Define regular expressions for log parsing (keep these as they are)
+SYSTEM_ROLE_LOG_RE = re.compile(
+    r"/SYSTEM-ROLE-(?P<role>[a-z0-9_]+)_(?P<test_name>tests_[a-z0-9_]+"
+    r"[.]yml)-.*-ANSIBLE-(?P<ansible_ver>[0-9.]+).*[.]log$"
+)
+SYSTEM_ROLE_TF_LOG_RE = re.compile(
+    r"/data/(?P<role>[a-z0-9_]+)-(?P<test_name>tests_[a-z0-9_]+)"
+    r"-ANSIBLE-(?P<ansible_ver>[0-9.]+)-(?P<tf_job_name>[0-9a-z_]+)"
+    r"-(?P<test_status>SUCCESS|FAIL)[.]log$"
+)
+
+
+async def get_file_data(log_url, timeout=30):
+    """Fetch file content from URL asynchronously."""
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.get(log_url, timeout=timeout) as response:
+                response.raise_for_status()
+                text = await response.text()
+                return text.splitlines()
+    except aiohttp.ClientError as e:
+        _logger.error("Failed to fetch log file: %s", str(e))
+        raise ValueError(f"Error fetching log file: {str(e)}")
+
+
+async def analyze_ansible_log(log_url, role_filter=["ALL"]):
+    """
+    Async task to analyze an Ansible log file for errors.
+
+    Args:
+        log_url (str): URL of the Ansible log file
+        role_filter (list): List of roles to filter by, or ["ALL"] for all roles
+
+    Returns:
+        dict: Information about the stored report
+    """
+    errors = await get_errors_from_ansible_log(role_filter, log_url)
+
+    # Store results in database
+    report = await sync_to_async(AnsibleLogReport.objects.create)(
+        log_url=log_url,
+        errors=errors,
+        error_count=len(errors),
+        role_filter=role_filter
+    )
```

Comment:
At this point you should add a `CreatedResource` which is a Tasking field / concept where we report to the user what a task made. Without this, the user only would have the URL and would have to requery the API to find the `AnsibleLogReport`.